### PR TITLE
Migrate another big chunk of the engine repo to `package:test`

### PR DESCRIPTION
**Pull request by matanlurey**
There are almost no behavioral changes.

Because `dart test` runs with assertions, and the former command does not, I had to tweak some of the engine tool tests because they would assert that there were duplicate names - but I didn't change any actual code besides the test expectations themselves/the fixtures.

This is not all of the engine, but is approximately 1/3 of all imports of `package:litetest` migrated.

### CODE REVIEW

File: tools/engine_tool/test/build_command_test.dart [hunk 1]
```diff
@@ -53,8 +52,9 @@ void main() {
       cannedProcesses: cannedProcesses,
     );
     try {
-      final List<Build> result = runnableBuilds(testEnv.environment, configs, true);
-      expect(result.length, equals(8));
+      final List<Build> result =
+          runnableBuilds(testEnv.environment, configs, true);
+      expect(result.length, equals(4));
```
Comment (loic-sharma) [inline tools/engine_tool/test/build_command_test.dart:57]: Could you expand why this went down to 4? (I'll keep reviewing, maybe it'll become obvious later in review)

Comment (matanlurey) [reply tools/engine_tool/test/build_command_test.dart:57]: Screenshot 2024-08-28 at 4 24 09 PM

This change (I don't think we cared about duplicate builds here, it was just copy pasta-d)

**Review by loic-sharma — APPROVED** [review]
Awesome!!!

File: tools/engine_tool/test/build_command_test.dart (hunk 1 above)
Comment (loic-sharma) [reply tools/engine_tool/test/build_command_test.dart:57]: Ah nice!

**Pull request merged**